### PR TITLE
feat: add namespace targetting -- ModRules in kubemod-system can now

### PR DESCRIPTION
**Pull request by jamiecore**
target any namespace
https://github.com/kubemod/kubemod/issues/51

### CODE REVIEW

**Comment by jamiecore** [comment]
pushin early to see if im on right track -- never used this workflow / unsure of any codegen stuffs -- adding tests after dinner

File: core/modrulestore.go [hunk 1]
```diff
@@ -121,11 +121,21 @@ func (s *ModRuleStore) Delete(namespace string, name string) {
...
 	modRules := []*ModRuleStoreItem{}
 
-	if namespaceModRules, ok := s.modRuleListMap[namespace]; ok {
-		for _, mrsi := range namespaceModRules {
-			if mrsi.modRule.Spec.Type == modRuleType && mrsi.IsMatch(jsonv) {
-				modRules = append(modRules, mrsi)
-			}
+	potentialRules := []*ModRuleStoreItem{}
+	for _, mrsi := range s.modRuleListMap[""] {
+		if namespace == "" && mrsi.compiledTargetNamespaceRegex == nil ||
```
Comment (vassilvk) [inline core/modrulestore.go:126]: Please replace with `if (namespace == "" && mrsi.compiledTargetNamespaceRegex == nil) || (mrsi.compiledTargetNamespaceRegex != nil && mrsi.compiledTargetNamespaceRegex.Match([]byte(namespace)))`, using parenthesis to demonstrate explicit intent.

File: core/modrulestore_test.go [hunk 2]
```diff
@@ -134,6 +138,8 @@ var _ = Describe("ModRuleStore", func() {
 		Entry("patch-10 on deployment-3 should work as expected", []string{"patch/patch-10.yaml"}, "deployment-3.json", "patch-10-deployment-3.txt"),
 		Entry("patch-11 on pod-6 should work as expected", []string{"patch/patch-11.yaml"}, "pod-6.json", "patch-11-pod-6.txt"),
 		Entry("patch-12 on pod-6 should work as expected", []string{"patch/patch-12.yaml"}, "pod-6.json", "patch-12-pod-6.txt"),
+		Entry("patch-13 on deployment-1 should work - ModRule in kubemod-system, but targetNamespaceRegex matches", []string{"patch/patch-13.yaml"}, "deployment-2.json", "patch-13-deployment-2.txt"),
+		Entry("patch-13b on deployment-1 should not work - ModRule in kubemod-system, but no targetNamespaceRegex", []string{"patch/patch-13b.yaml"}, "deployment-2.json", "empty-array.txt"),
```
Comment (vassilvk) [inline core/modrulestore_test.go:142]: Please rename `patch-13b.yaml` to `patch-14.yaml`.

File: core/testdata/modrules/patch/patch-13.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+apiVersion: api.kubemod.io/v1beta1
+kind: ModRule
+metadata:
+  name: modrule-1
+  namespace: kubemod-system
+spec:
+  type: Patch
+  targetNamespaceRegex: '\w{2,}-[^ ]{3}[[:lower:]]+?[[:xdigit:]]+'
+  # ^^^ regex to match 'my-namespace' -- the namespace used by the test suite
```
Comment (vassilvk) [inline core/testdata/modrules/patch/patch-13.yaml:9]: Please move comment above the line to make it consistent with the rest of the code.

**Review by vassilvk — CHANGES_REQUESTED** [review]
Thank you, @jamiecore, looks good apart from a a few minor changes I noted in my review.

File: core/testdata/modrules/patch/patch-13.yaml (hunk 3 above)
Comment (jamiecore) [reply core/testdata/modrules/patch/patch-13.yaml]: 👍

File: core/modrulestore_test.go (hunk 2 above)
Comment (jamiecore) [reply core/modrulestore_test.go]: 👍

File: core/modrulestore.go (hunk 1 above)
Comment (jamiecore) [reply core/modrulestore.go]: 👍

**Review by vassilvk — APPROVED** [review]
Thank you!

**Comment by vassilvk** [comment]
Thank you @jamiecore!

**Pull request merged**